common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java

```diff
@@ -905,6 +905,26 @@ public List<ICacheRecord> getAccountsFromResultBundle(@NonNull final Bundle bund
         return JsonExtensions.getICacheRecordListFromJsonString(accountJson);
     }
 
+    /**
+     * Get resource account record from the result bundle. If successful, new account
+     * record part of ICachedRecord is returned.
+     * @param bundle The result bundle from the broker.
+     * @throws BaseException
+     */
+    public ICacheRecord resourceAccountRecordFromBundle(@NonNull final Bundle bundle) throws BaseException {
```

Comment:
Do we need this method? Could the caller just call the `getAccountsFromResultBundle`?

Reply:
It wraps additional logic. I am checking if 0 or more than 1 accounts are returned.

Reply:
Yes, but the way I see it the method is too generic (which is nice) and essentially is a wrapper to get just the first account record from cache. IIRC, we are doing something similar for SingleAccountMode PCA. At some point we probably should consolidate into 1 generic method.

Reply:
Also wondering if rather than returning the bundle and then letting MSAL validate it if Broker itself should be throwing in the first place?

Reply:
Yes, there will be check on broker side.. I don't have implementation of the broker operation right now, but it will ensure to return on the one account object that it just created.
Correspondingly the MSAL side impl may also change. the method signature `public ICacheRecord resourceAccountRecordFromBundle(@NonNull final Bundle bundle)` remains same